example/app/example-app/projections.go

```diff
@@ -0,0 +1,352 @@
+	}
+
+	var cosmosAppClient cosmosapp.Client
+	if config.CosmosApp.Insecure {
+		cosmosAppClient = cosmosapp_infrastructure.NewInsecureHTTPClient(
+			config.CosmosApp.HTTPRPCUrl, config.Blockchain.BondingDenom,
+		)
+	} else {
+		cosmosAppClient = cosmosapp_infrastructure.NewHTTPClient(
+			config.CosmosApp.HTTPRPCUrl, config.Blockchain.BondingDenom,
+		)
+	}
+
+	projections := make([]projection_entity.Projection, 0, len(config.Projection.Enables))
+	initParams := InitProjectionParams{
+		Logger:  logger,
+		RdbConn: rdbConn,
+
+		CosmosAppClient:       cosmosAppClient,
+		AccountAddressPrefix:  config.Blockchain.AccountAddressPrefix,
+		ConsNodeAddressPrefix: config.Blockchain.ConNodeAddressPrefix,
+
+		GithubAPIUser:    config.GithubAPI.Username,
+		GithubAPIToken:   config.GithubAPI.Token,
+		MigrationRepoRef: config.GithubAPI.MigrationRepoRef,
+
+		ServerMigrationRepoRef: customConfig.ServerGithubAPI.MigrationRepoRef,
+	}
+	for _, projectionName := range config.Projection.Enables {
+		projection := InitProjection(
+			projectionName, initParams,
+		)
+		if projection == nil {
+			continue
+		} else {
+			if onInitErr := projection.OnInit(); onInitErr != nil {
+				logger.Errorf(
+					"error initializing projection %s, system will attempt to initialize the projection again on next restart: %v",
+					projection.Id(), onInitErr,
+				)
+				continue
+			}
+		}
+		projections = append(projections, projection)
+	}
+
+	// Append additional projection
+	for _, projectionName := range config.Projection.Enables {
+		projection := InitAdditionalProjection(
+			projectionName, initParams,
+		)
+		if projection == nil {
+			continue
+		} else {
+			if onInitErr := projection.OnInit(); onInitErr != nil {
+				logger.Errorf(
+					"error initializing additional projection %s, system will attempt to initialize the projection again on next restart: %v",
+					projection.Id(), onInitErr,
+				)
+				continue
+			}
+		}
+		projections = append(projections, projection)
+	}
+
+	logger.Infof("Enabled the follow projection: [%s]", strings.Join(config.Projection.Enables, ", "))
+
+	return projections
+}
+
+func InitAdditionalProjection(name string, params InitProjectionParams) projection_entity.Projection {
+	connString := params.RdbConn.(*pg.PgxConn).ConnString()
+
+	githubMigrationHelperConfigForCustomProjection := github_migrationhelper.Config{
+		GithubAPIUser:    params.GithubAPIUser,
+		GithubAPIToken:   params.GithubAPIToken,
+		MigrationRepoRef: params.ServerMigrationRepoRef,
+		ConnString:       connString,
+	}
+
+	switch name {
+	case "Example":
+		sourceURL := generateGithubMigrationSrouceURLForCustomProjection(name, githubMigrationHelperConfigForCustomProjection)
+		databaseURL := migrationhelper.GenerateDefaultDatabaseURL(name, connString)
+		migrationHelper := github_migrationhelper.NewGithubMigrationHelper(sourceURL, databaseURL)
+
+		return example.NewAdditionalProjection(params.Logger, params.RdbConn, migrationHelper)
+	}
+
+	return nil
```

Comment:
I am thinking if we should panic here instead?

Reply:
Reason why not using panic here is that we separated projections initialisation function into two (provided one + additional one) and this way could make the customisation more clear for devs.

Reply:
I think panic is better here. As we can give a more explicit error message to the user, `Man, you input an undefined projection name` 😂  It could help the user save sometimes for debugging his own typo.

Reply:
As devs using the config file provided in this example, they are not likely to have typo for projection names. If we use panic, it will mean we will combining the initialisations together. Not a very good way to show devs which part is customisation projection and which part is provided projection. Do you think it is still worth to do it?

Reply:
I see. Yea, that also make sense. I am ok with the current design.

Reply:
@calvinaco do you agree?

Reply:
Yes I think as an example app this is fine.

In a more production setup I will prefer to use panic because typo will happens (as always haha :)) and I want to avoid that error being swallowed.